app/screens/Product/Product.js

```diff
@@ -3,26 +3,33 @@ import { Text, View, Image, TouchableHighlight } from 'react-native';
 
 import { iconImages } from './../../images';
 
-import { styles } from './styles';
+import { styles, buttonUnderlayColor } from './styles';
 
 class Product extends Component {
+    redirect() {
+        this.props.navigation.navigate('ProductList', {
+            screen: 'ProductList'
+        });
+    }
+
     render() {
+        const {navigation} = this.props,
+            title = navigation.getParam('title'),
```

Comment:
If params is not changing during render you can extract title, image and description just once in constructor.